### DISCUSSION TITLE
Connecting to an iDraw plotter from UGS on Linux

### DISCUSSION THREAD

**Discussion by snow6oy** [winder/Universal-G-Code-Sender#2589] [Help] [answered]
Hi,

I am having problems connecting to an idraw plotter from UGS. Firstly I downloaded the Linux version from.
https://universalgcodesender.com/download/

After installation I saw this in the messages.log
```

>Log Session: Saturday, August 10, 2024 at 11:39:47 PM Central European Summer Time
>System Info:
  Product Version         = Universal Gcode Sender 18-7d33005615b3c744564f3d38ab9ef14bf8c3ec44
  Operating System        = Linux version 6.5.0-45-generic running on amd64
  Java; VM; Vendor        = 17.0.8.1; OpenJDK 64-Bit Server VM 17.0.8.1+1; Eclipse Adoptium
  Runtime                 = OpenJDK Runtime Environment 17.0.8.1+1
  Java Home               = /home/gavin/Downloads/linux-x64-ugs-platform-app-2.1.8/ugsplatform-linux-x64/jdk/jdk-17.0.8.1+1-jre
  System Locale; Encoding = en_GB (ugsplatform); UTF-8

```
The I grabbed a test file.

https://github.com/winder/Universal-G-Code-Sender/blob/master/test_files/spiral.gcode

The file didn't show in Visualizer but I went ahead anyway to try and connect. In the GUI console I saw

```
*** Connecting to jserialcomm://ttyACM0:115200
OK
*** Fetching device status
>>> ?
!8 Err: Unknown command '!1:2131'
!7 Err: Extra parmater
*** Fetching device status (2 of 10)...
>>> ?
!8 Err: Unknown command '?:3F'
*** Fetching device status (3 of 10)...

```
And in the messages.log it was written

```
INFO [com.willwinder.universalgcodesender.model.GUIBackend]: Setting gcode file. /home/gavin/code/gcode-plotter/spiral.gcode
INFO [com.willwinder.universalgcodesender.model.GUIBackend]: Applying new command processor RunFromProcessor
INFO [com.willwinder.universalgcodesender.model.GUIBackend]: Start preprocessing
INFO [com.willwinder.universalgcodesender.model.GUIBackend]: Preprocessing /home/gavin/code/gcode-plotter/spiral.gcode to /tmp/10712442086714936298/spiral.gcode_ugs_1723490034659
INFO [com.willwinder.universalgcodesender.model.GUIBackend]: Took 17ms to preprocess
INFO [com.willwinder.ugs.nbm.visualizer.renderables.GcodeModel]: About to process {}
INFO [com.willwinder.ugs.nbm.visualizer.renderables.GcodeModel]: Object bounds: X (-40.0, 40.0)
INFO [com.willwinder.ugs.nbm.visualizer.renderables.GcodeModel]:                Y (-19.99604659540131, 19.99604659540131)
INFO [com.willwinder.ugs.nbm.visualizer.renderables.GcodeModel]:                Z (-20.0, 20.0)
INFO [com.willwinder.ugs.nbm.visualizer.renderables.GcodeModel]: Center = com.willwinder.universalgcodesender.model.Position@69787f72[units=UNKNOWN,a=NaN,b=NaN,c=NaN,x=0.0,y=0.0,z=0.0]
INFO [com.willwinder.ugs.nbm.visualizer.renderables.GcodeModel]: Num Line Segments :957
INFO [com.willwinder.ugs.nbm.visualizer.renderables.GcodeModel]: Done setting gcode file.
INFO [com.willwinder.universalgcodesender.communicator.AbstractCommunicator]: Connecting to controller using class: JSerialCommConnection with url jserialcomm://ttyACM0:115200
SEVERE [UGSEventDispatcher]: Could not dispatch the event ControllerStatusEvent to the listener OverridesPanel

```
The plotter is visible to the OS.

```
$ lsusb |grep EiBot
Bus 003 Device 005: ID 04d8:fd92 Microchip Technology, Inc. EiBotBoard,

```
Using the Inkscape extension for idraw I can connect and plot. However, my use-case requires sending gcode directly to the idraw machine.

Any ideas how I can debug this further?

Thanks
Gavin

**Comment by breiler** [comment]
iDraw uses a protocol that is not supported by UGS. To my knowledge it doesn't work with gcode. 

Have you read in some documentation that it is supposed to work?

**Comment by snow6oy** [comment]
Thanks for a quick response. I bought my iDraw from Uunatek. On their home page it says
> Although UUNA TEK® iDraw is commonly controlled through a set of extensions to Inkscape, there are alternative methods of plotting using the command-line interface (CLI), Python, or GRBL command.

And they have [this article about using UGC with idraw](https://uunatek.com/blogs/tips-and-tricks/understanding-gcode-and-setting-up-the-idraw-machine-for-the-first-time-gcode-usage). 

But I think you may have a point because I see this uniquely spelt error 'Extra parmater' when I have tried to use other software to send gcode to the iDraw. Now I am worried that the firmware has some non-standard gcode implementation :cry:

I will investigate further.

Gavin

**Reply by breiler** [reply, marked as answer]
It might support gcode, I have no idea. But given the responses you get when connecting to it shows that the protocol for transferring the gcode to the controller is different than with GRBL based firmwares. So you will have to ask UUNA TEK about this.

**Reply by snow6oy** [reply]
UUNA TEK confirmed that iDraw 1.0 is based on EBB firmware, not GRBL. Since then I found an explanation in this discussion about [Gcode support](https://github.com/techninja/cncserver/issues/44). In case anyone needs to check the following worked on my iDraw. Firslty connect to the plotter using tio
`$ tio --socket unix:/tmp/tio-socket0 /dev/ttyACM0`
Then in another terminal send the version command
`echo V | nc -UN /tmp/tio-socket0`
In my case the reply was
`EBBv13_and_above EB Firmware Version 2.8.1`
